src/main/java/org/opentripplanner/routing/algorithm/raptoradapter/transit/Transfer.java

```diff
@@ -60,6 +80,11 @@ public List<Edge> getEdges() {
   }
 
   public Optional<RaptorTransfer> asRaptorTransfer(StreetSearchRequest request) {
+    return toRaptor(request)
+      .filter(s -> s.generalizedCost() < MAX_TRANSFER_RAPTOR_COST && s.generalizedCost() >= 0);
+  }
+
+  private Optional<RaptorTransfer> toRaptor(StreetSearchRequest request) {
```

Comment:
There is a problem with level of abstraction here. The check is not the most imortanet thing - it is just a small detail that should be stuffed away. It makes reading this code difficult. Consider moving the check to line 92:
```
92: int cost = costLimitSanityCheck(durationSeconds * walkPreferences.reluctance());
:
95:   RaptorCostConverter.toRaptorCost(cost),
---
/** <Put implementation details here, move doc from MAX_TRANSFER_RAPTOR_COST to here> */ 
private static int costLimitSanityCheck(int cost) {
  return (cost >= 0 || cost <= MAX_TRANSFER_COST) ? cost : MAX_TRANSFER_COST;
}
```
Set the max cost to a value, not an expression of MAX_INT. I think 2_000_000 is a good value - this looks like it is set by man, not computed and it allow 10 transfers of this kind before an overflow. A cost of 23 days should be enough to rule out this transfer. Note! I am not dropping the transfer, I am just setting an upper bound. It might be good to log here as well, encouraging fixing the high cost reluctance to be reconsidered.

If you want, you can compute the MAX like this: `RaptorCostConverter.toOtpDomainCost(RaptorConstants.UNREACHED_HIGH / 10)` should be `2_000_000`.